src/pack-objects.h

```diff
@@ -97,5 +97,7 @@ struct git_packbuilder {
 };
 
 int git_packbuilder__write_buf(git_str *buf, git_packbuilder *pb);
+int prepare_pack(git_packbuilder *pb);
```

Comment:
I don't think that we need to expose this publicly (yet) - I agree that exposing APIs that aren't "needed" gives us a maintenance burden moving forward.  But let's give it a nicer name even if it's an internal API.

```suggestion
int git_packbuilder__prepare(git_packbuilder *pb);
```